kube-dns deployment was silently missing

I'd been running the E2E test against #465 and it failed like:

```
Summarizing 4 Failures:

[Fail] [k8s.io] DNS [It] should provide DNS for services [Conformance]
/go/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/dns.go:219

[Fail] [k8s.io] Networking [It] should provide Internet connection for containers [Conformance]
/go/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/networking.go:48

[Fail] [k8s.io] DNS [It] should provide DNS for the cluster [Conformance]
/go/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/dns.go:219

[Fail] [k8s.io] Kubectl client [k8s.io] Guestbook application [It] should create and stop a working application [Conformance]
/go/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:1580

Ran 117 of 474 Specs in 3522.154 seconds
FAIL! -- 113 Passed | 4 Failed | 0 Pending | 357 Skipped --- FAIL: TestE2E (3522.17s)
FAIL
```

Looking into the kube-system namespace revealed that kube-dns pods were completely missing.

```
$ KUBECONFIG=assets/kubeawstest3/kubeconfig kubectl get po --all-namespaces
NAMESPACE     NAME                                                                    READY     STATUS    RESTARTS   AGE
kube-system   heapster-v1.3.0-567306696-1qcdm                                         2/2       Running   0          1h
kube-system   kube-apiserver-ip-10-0-0-124.ap-northeast-1.compute.internal            1/1       Running   0          1h
kube-system   kube-controller-manager-ip-10-0-0-124.ap-northeast-1.compute.internal   1/1       Running   0          1h
kube-system   kube-dns-autoscaler-2813114833-ctfsp                                    1/1       Running   0          1h
kube-system   kube-proxy-ip-10-0-0-124.ap-northeast-1.compute.internal                1/1       Running   0          1h
kube-system   kube-proxy-ip-10-0-0-135.ap-northeast-1.compute.internal                1/1       Running   0          1h
kube-system   kube-proxy-ip-10-0-0-206.ap-northeast-1.compute.internal                1/1       Running   0          1h
kube-system   kube-proxy-ip-10-0-0-36.ap-northeast-1.compute.internal                 1/1       Running   0          1h
kube-system   kube-proxy-ip-10-0-0-61.ap-northeast-1.compute.internal                 1/1       Running   0          1h
kube-system   kube-proxy-ip-10-0-0-78.ap-northeast-1.compute.internal                 1/1       Running   0          1h
kube-system   kube-scheduler-ip-10-0-0-124.ap-northeast-1.compute.internal            1/1       Running   0          1h
kube-system   kubernetes-dashboard-v1.5.1-0q4n9                                       1/1       Running   0          1h
```

[Manually re-running `/opt/bin/install-kube-system`](https://gist.github.com/mumoshu/e4bfa66bbc6f81ced77a8a4ec1b54fa3#file-install-kube-system-2-rerun-log-L1-L227) did fix the problem.

```
$ FOCUS='.*DNS.*Conformance.*' ETCD_COUNT=3 KUBE_AWS_CLUSTER_NAME=kubeawstest3 ./run conformance
...snip...
Ran 2 of 474 Specs in 33.560 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 472 Skipped PASS
```

```
$ FOCUS='.*Networking.*Internet.*Conformance.*' ETCD_COUNT=3 KUBE_AWS_CLUSTER_NAME=kubeawstest3 ./run conformance
...snip...
Ran 1 of 474 Specs in 7.361 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 473 Skipped PASS
```

```
$ FOCUS='.*Kubectl.*Guestbook.*Conformance.*' ETCD_COUNT=3 KUBE_AWS_CLUSTER_NAME=kubeawstest3 ./run conformance
...snip...
Ran 1 of 474 Specs in 96.491 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 473 Skipped PASS
```

How could it happen?
Looking into `journalctl -u install-kube-system` carefully showed that in the very early stage of the script, the first curl command for installing `kube-dns` had failed due to the missing `kube-system` namespace!

https://gist.github.com/mumoshu/e4bfa66bbc6f81ced77a8a4ec1b54fa3#file-install-kube-system-1-boot-log-L130

So let's fix it; should we wait until the `kube-system` namespace to be available in the beginning of `install-kube-system` or elsewhere?

--- a/core/controlplane/config/templates/cloud-config-controller
+++ b/core/controlplane/config/templates/cloud-config-controller
@@ -500,6 +500,11 @@ write_files:
           /usr/bin/docker run --rm --net=host -v /srv/kubernetes:/srv/kubernetes {{.HyperkubeImage.RepoWithTag}} /hyperkube kubectl "$@"
       }
 
+      while ! kubectl get ns kube-system; do
+        echo Waiting until kube-system created.
+        sleep 3
+      done
+
       mfdir=/srv/kubernetes/manifests
 
       {{ if .UseCalico }}

